src/Orchard.Environment.Shell.Data/ShellDescriptorManager.cs

```diff
@@ -31,14 +32,20 @@ public class ShellDescriptorManager : IShellDescriptorManager
             _eventBus = eventBus;
             _session = session;
             _logger = logger;
+
+            _asyncLazyShellDescriptor = new Lazy<Task<ShellDescriptor>>(() => Task.Run(() =>
```

Comment:
I don't really like it. What triggers the issue? Where is the parallelism that creates a race condition?